### PR TITLE
adding csr support to kmeans 

### PR DESCRIPTION
**Pull request by RomaA2000**

### CODE REVIEW

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 1]
```diff
@@ -77,20 +77,23 @@ def _daal4py_compute_starting_centroids(X, X_fptype, nClusters, cluster_centers_
 
     def is_string(s, target_str):
         return isinstance(s, string_types) and s == target_str
-
+    is_sparse = sp.isspmatrix(X)
+    default_method = "lloydCSR" if is_sparse else "defaultDense"
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:81]: I think it's better to move each method to its own branch.

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 2]
```diff
@@ -137,12 +140,16 @@ def _daal4py_kmeans_compatibility(nClusters, maxIterations, fptype = "double",
 
 def _daal4py_k_means_predict(X, nClusters, centroids, resultsToEvaluate = 'computeAssignments'):
     X_fptype = getFPType(X)
+    is_sparse = sp.isspmatrix(X)
+    method = "lloydCSR" if is_sparse else "defaultDense"
+    print("X type predict: ", type(X))
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:145]: don't forget to delete it

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 3]
```diff
@@ -155,17 +162,19 @@ def _daal4py_k_means_fit(X, nClusters, numIterations, tol, cluster_centers_0, n_
 
     X_fptype = getFPType(X)
     abs_tol = _tolerance(X, tol) # tol is relative tolerance
-
+    print("X type fit: ", type(X))
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:165]: don't forget to delete it

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 4]
```diff
@@ -155,17 +162,19 @@ def _daal4py_k_means_fit(X, nClusters, numIterations, tol, cluster_centers_0, n_
...
+    is_sparse = sp.isspmatrix(X)
+    method = "lloydCSR" if is_sparse else "defaultDense"
     best_inertia, best_cluster_centers = None, None
     best_n_iter = -1
-
+    print("method fit: ", method)
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:170]: don't forget to delete it

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 5]
```diff
@@ -246,9 +255,9 @@ def fit(self, X, y=None, sample_weight=None):
                 f"max_iter should be > 0, got {self.max_iter} instead.")
 
     # avoid forcing order when copy_x=False
-    order = "C" if self.copy_x else None
-    X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32],
-                    order=order, copy=self.copy_x)
+    # order = "C" if self.copy_x else None
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:258]: delete it?

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 6]
```diff
@@ -264,9 +273,8 @@ def fit(self, X, y=None, sample_weight=None):
                          " {}".format(str(algorithm)))
 
 
-    daal_ready = not sp.issparse(X)
-    daal_ready = daal_ready and hasattr(X, '__array__')
-
+    daal_ready = True
+    print("X type: ", type(X))
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:277]: don't forget to delete it

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 7]
```diff
@@ -345,6 +352,7 @@ def __init__(self, n_clusters=8, *, init='k-means++', n_init=10,
             copy_x=copy_x, n_jobs=n_jobs, algorithm=algorithm)
 
     def fit(self, X, y=None, sample_weight=None):
+        print("We are in daal4py")
```
Comment (PetrovKP) [inline daal4py/sklearn/cluster/_k_means_0_23.py:355]: don't forget to delete it

**Comment by PetrovKP** [comment]
@owerbat , @Alexsandruss please, review

File: daal4py/sklearn/cluster/_k_means_0_23.py [hunk 8]
```diff
@@ -246,9 +252,9 @@ def fit(self, X, y=None, sample_weight=None):
                 f"max_iter should be > 0, got {self.max_iter} instead.")
 
     # avoid forcing order when copy_x=False
-    order = "C" if self.copy_x else None
-    X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32],
-                    order=order, copy=self.copy_x)
+    # order = "C" if self.copy_x else None
+    # X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32],
```
Comment (Pahandrovich) [inline daal4py/sklearn/cluster/_k_means_0_23.py:256]: Do you need these comments?

**Review by Pahandrovich — APPROVED** [review]

**Review by owerbat — APPROVED** [review]
👍

**Pull request merged**